### PR TITLE
Improve error when using => on object literal method

### PR DESCRIPTION
**Pull request by singalhimanshu**
The following code:
    let o = {
      method() => {
      },
    };

Should report error_methods_should_not_have_arrow_operator.
Fix: #292

### CODE REVIEW

File: src/quick-lint-js/parse.h [hunk 1]
```diff
@@ -1263,6 +1263,14 @@ class parser {
       }
       this->skip();
 
+      if (this->peek().type == token_type::equal_greater) {
```
Comment (strager) [inline src/quick-lint-js/parse.h:1266]: This applies to `function f() => {}` too, right? If so, we should add a test case for that syntax. And we should tweak the message. (Perhaps have two different messages? One saying "functions" and one saying "methods".)

Same with `class C { f() => {} }`, except those are called methods, so the current wording works.

**Review by strager — APPROVED** [review]
Looks good!

File: src/quick-lint-js/parse.h (hunk 1 above)
Comment (singalhimanshu) [reply src/quick-lint-js/parse.h:1266]: > This applies to `function f() => {}` too, right?

Right



> Perhaps have two different messages? One saying "functions" and one saying "methods

Do you mean two different errors? Also how do we know if we are in a function or method?

Comment (strager) [reply src/quick-lint-js/parse.h:1266]: > Do you mean two different errors?

Yes, or a single error type with a parameterized message. I'm leaining toward two different error types.

> Also how do we know if we are in a function or method?

Dunno.

**Review by strager — APPROVED** [review]

**Pull request merged**